scripts/create_kdu_compress_encode_scripts.py

```diff
@@ -85,11 +85,11 @@
   elif encoding_parameter_set == 4:
     # HT digital bodelian - lossless - https://image-processing.readthedocs.io/en/latest/jp2_profile.html#kduusage
     encoding_parameters="\"Cprecincts={256,256},{256,256},{128,128}\" \"Stiles={512,512}\" Corder=RPCL ORGgen_plt=yes ORGtparts=R \"Cblk={64,64}\" Cuse_sop=yes Cuse_eph=yes -flush_period 1024 Creversible=yes Cmodes=HT -rate -"
```

Comment:
This still has a rate of - is that correct?

Reply:
yes, with kakadu's kdu_compress tool `-rate -` means no rate limit, which is what happens with lossless compression.  This is how the original bodelian commands are documented [1] so I included it for consistency.  With recent versions of kakadu, if you use kakadu's kdu_compress `Creversible=yes` argument you don't actually need to use `-rate -` argument. 

The rate of lossless compression is dependent on the image characteristics.  Easy to compress images have lower lossless data rates and difficult to compress images have higher data rate limits.  Using `-rate -` is usually useful when creating multiple quality layers within a single JPEG 2000 codestream, but in IIFF we don't use multiple quality layers.  Multiple quality layers can be useful in some other applications when the decoding client is accessing the codestream directly through bandwidth limited channels.  In IIIF, the server has very high bandwidth access to the source file. To make multiple quality layers with specified rate targets in each layer you could use something like `-rate 1,2,3,4` which would perform lossy compression using 4 quality layers and set the bits per pixel rate target in each quality to 1, 2, 3 and 4 bits per pixel respectively.  If you want the last quality layer (the highest quality) to not have any rate limit for lossless compression, instead of being limited to 4 bits per pixel, the last argument could be replaced with `-` like this `-rate 1,2,3,-`

[1] https://image-processing.readthedocs.io/en/latest/jp2_profile.html